src/test/java/org/cactoos/iterable/MatchedTest.java

```diff
@@ -155,4 +157,40 @@ public void iterablesOfDifferentTypes() {
         ).affirm();
     }
 
+    @Test
+    public void traversableSeveralTimes() {
```

Comment:
@andreoss could you explain what this test is actually doing, it seems very complex if the objective is only to see if `Matched` can be traversed multiple times: for this you only need to test that it is equal to the expected content twice with `AllOf` matcher for example… If we can avoid relying on too many diverse complex objects such as `RangeOf`, it would be best too, because I wonder if we are testing `RangeOf` here or `Matched`.

Reply:
@victornoel I removed the second test, because it was doing the same thing as the first one. 
The objective here is to check that `iterator` is not empty after all elements have been accessed. `RangeOf` here was just to create an `Iterable`, I've replaced it with plain `IterableOf`